cg/cli/upload/base.py

```diff
@@ -57,29 +58,29 @@ def upload(context: click.Context, case_id: str | None, restart: bool):
     """Upload results from analyses"""
 
     config_object: CGConfig = context.obj
-    upload_api: UploadAPI = MipDNAUploadAPI(config=config_object)  # default upload API
+    upload_api: UploadAPI = MipDNAUploadAPI(config_object)
 
     LOG.info("----------------- UPLOAD -----------------")
 
     if context.invoked_subcommand is not None:
         context.obj.meta_apis["upload_api"] = upload_api
     elif case_id:  # Provided case ID without a subcommand: upload everything
         try:
-            upload_api.analysis_api.status_db.verify_case_exists(case_internal_id=case_id)
-            case: Case = upload_api.status_db.get_case_by_internal_id(internal_id=case_id)
+            upload_api.analysis_api.status_db.verify_case_exists(case_id)
+            case: Case = upload_api.status_db.get_case_by_internal_id(case_id)
             upload_api.verify_analysis_upload(case_obj=case, restart=restart)
         except AnalysisAlreadyUploadedError:
             # Analysis being uploaded or it has been already uploaded
             return
 
-        # Update the upload API based on the data analysis type (MIP-DNA by default)
-        # Upload for balsamic, balsamic-umi and balsamic-qc
         if Pipeline.BALSAMIC in case.data_analysis:
-            upload_api = BalsamicUploadAPI(config=config_object)
+            upload_api = BalsamicUploadAPI(config_object)
         elif case.data_analysis == Pipeline.RNAFUSION:
-            upload_api = RnafusionUploadAPI(config=config_object)
+            upload_api = RnafusionUploadAPI(config_object)
         elif case.data_analysis == Pipeline.MIP_RNA:
-            upload_api: UploadAPI = MipRNAUploadAPI(config=config_object)
+            upload_api = MipRNAUploadAPI(config_object)
+        elif case.data_analysis == Pipeline.MICROSALT:
+            upload_api = MicrosaltUploadAPI(config_object)
```

Comment:
Consider breaking out into a small utils function, e.g. `get_upload_api_for_pipeline`

Reply:
Agree, I think the module needs refactoring but I'll leave it for now